src/utils/cache.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {map} from './object';
+
+
+/**
+ * An LRU Cache
+ * @template T
+ */
+export class Cache {
+  /**
+   * Construct a new Cache with the given size
+   * @param size {number=}
```

Comment:
The type definitions in this file are backwards. Should be `@param {number=} size`